### PR TITLE
Port-Group speed updates and Interface valid-speeds issue. #67

### LINKED ISSUE #67: Port-Group speed updates and Interface valid-speeds issue.
If port group speed is set to 25G , all interface in the port-group are updated to have the speed 25G, but the valid speeds and adv-speed set is still 1G and 10G which is not correct.

- [x] The correct behaviour would be if port-group speed is updated, after receving the port-group update via gNMI subscription make an additional gnmi get to update the valid_speeds field for interface at sonic-port:sonic-port/PORT/PORT_LIST={interface_name}.
- [x] Write test case for above.

### PR DESCRIPTION
**Pull request by rahul-1475**
Fixes #67 

issue: 

If port group speed is set to 25G , all interface in the port-group are updated to have the speed 25G, but the valid speeds and adv-speed set is still 1G and 10G which is not correct.

- The correct behaviour would be if port-group speed is updated, after receving the port-group update via gNMI subscription make an additional gnmi get to update the valid_speeds field for interface at sonic-port:sonic-port/PORT/PORT_LIST={interface_name}.

- Write test case for above.

Fixes: 
- Added interface discovery on speed update.
- Added test cases to validate. valid speeds update.

### CODE REVIEW

**Comment by rahul-1475** [comment]
other pull request:

https://github.com/STORDIS/orca_backend/pull/42

File: pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tool.poetry]
-version = "1.3.15"
+version = "1.3.16"
```
Comment (kamalkrbh) [inline pyproject.toml:2]: @rahul-1475, Version should be +1 , similar in the backend as well.
```suggestion
version = "1.3.17"
```

File: orca_nw_lib/portgroup.py [hunk 2]
```diff
@@ -81,7 +81,7 @@ def get_port_group_members(device_ip: str, group_id):
             op_dict.append(mem_if.__properties__)
     return op_dict
 
-def get_port_group_of_interface(device_ip: str, if_name:str):
+def get_port_group_of_interface(device_ip: str, group_id:str):
```
Comment (kamalkrbh) [inline orca_nw_lib/portgroup.py:84]: @rahul-1475 , The purpose of this function is to know the port group of a given interface. So input parameters has to be interface name instead fix has to be in the body of the function where DB function get_port_group_of_if_from_db should be called.

**Review by kamalkrbh — CHANGES_REQUESTED** [review]
Have a look on above comments in the files.

File: orca_nw_lib/portgroup.py (hunk 2 above)
Comment (rahul-1475) [reply orca_nw_lib/portgroup.py]: - changed code get to use function get_port_group_of_if_from_db  to port group name by interface.
- created api to get prot group name by interface

File: pyproject.toml (hunk 1 above)
Comment (rahul-1475) [reply pyproject.toml]: changed orca_nw_lib version to 1.3.17

**Review by kamalkrbh — CHANGES_REQUESTED** [review]
@rahul-1475 , Please have a look at merge conflicts.

**Pull request merged**